txcache/accountBreadcrumb.go

```diff
@@ -38,3 +39,70 @@ func (breadcrumb *accountBreadcrumb) updateNonce(lastNonce core.OptionalUint64)
 		breadcrumb.lastNonce = lastNonce
 	}
 }
+
+func (breadcrumb *accountBreadcrumb) createOrUpdateVirtualRecord(
+	virtualAccountsByAddress map[string]*virtualAccountRecord,
+	accountState state.UserAccountHandler,
+	address string,
+) {
+	virtualRecord, ok := virtualAccountsByAddress[address]
+	if !ok {
+		initialBalance := accountState.GetBalance()
+		virtualRecord = newVirtualAccountRecord(breadcrumb.initialNonce, initialBalance)
+		virtualAccountsByAddress[address] = virtualRecord
+	}
+
+	virtualRecord.updateVirtualRecord(breadcrumb)
+}
+
+func (breadcrumb *accountBreadcrumb) isContinuous(
```

Comment:
If you choose to apply the suggested refactoring using an ephemeral provider of selection sessions, this function would belong to that struct, as well.

Reply:
Maybe `isContinousWithRespectToNonces` or `followsNonceContinuity`, or `hasNonceGap` (negative check, reversed flow). Just an opinion, all right as it is now.